pkg/operator/operator.go

```diff
@@ -202,40 +203,53 @@ func (c *csiDriverOperator) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
-func (c *csiDriverOperator) sync() error {
-	instance, err := c.client.GetOperatorInstance()
-	if err != nil {
-		if errors.IsNotFound(err) {
-			klog.Warningf("Operator instance not found: %v", err)
-			return nil
-		}
-		return err
-	}
-
+func (c *csiDriverOperator) handleDeletion(instance *v1alpha1.AWSEBSDriver) (bool, error) {
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
 		// CR is NOT being deleted, so make sure it has the finalizer
 		if addFinalizer(instance, operatorFinalizer) {
 			c.client.UpdateFinalizers(instance)
 		}
+		return false, nil
+	}
+
+	// Deletion timestamp is set, so user tried to delete the CR
+	// First, set progressing condition to true because we're
+	// trying to transition the operand to a different state
+	c.setCondition(instance, operatorv1.OperatorStatusTypeProgressing, true, reasonRemovingOperand, "")
+	defer c.client.UpdateOperatorStatus(instance.ResourceVersion, &instance.Status.OperatorStatus)
```

Comment:
Why is it set via `defer`? You loose `UpdateOperatorStatus` error. `isCSIDriverInUse` below is very likely to succeed - it just lists internal cache. What about this:

```
	ok, err := c.isCSIDriverInUse()
￼	if err != nil {
￼		return true, err
￼	}

	if ok {
		c.setCondition(instance, operatorv1.OperatorStatusTypeProgressing, true, reasonRemovingOperand, msg)
		_, err =  c.client.UpdateOperatorStatus(instance.ResourceVersion, &instance.Status.OperatorStatus)
		// handle err

		...
	} else {
		c.setCondition(instance, operatorv1.OperatorStatusTypeProgressing, true, reasonRemovingOperand, someOtherMsg)
		_, err =  c.client.UpdateOperatorStatus(instance.ResourceVersion, &instance.Status.OperatorStatus)
		// handle err
		...
	}
```

Reply:
That's better, but I realized that we'd be updating the status (in the api server) multiple times in a single sync.

Another problem is that we're mutating the input (`instance`) instead of mutating a copy of it.

I re-worked the patch to address these issues.